### DISCUSSION TITLE
Support for the Media Session API

### DISCUSSION THREAD

**Discussion by n-3-0** [ionic-team/capacitor#3969] [Ideas]
Capacitor.js is a fantastic library and platform. However, as someone who is extremely new to it, there are some areas that could use improvement. One such place is media. Audio is a breeze using either the Web Audio API or a capacitor/cordova plugin for native audio. That said, media controls are not as straightforward.
The best available option at the moment, sans Ionic, is the [capacitor-music-controls-plugin](https://github.com/ingageco/capacitor-music-controls-plugin/) repo, which I was not able to get working using the documented installation.
I am attempting to port my React PWA music player to more native solutions so I can support local audio. This is not something traditionally feasible using a PWA, because access to local files is (justifiably) heavily-regulated. Capacitor.js provides this essential bridge. Between Capacitor.js and Electron, my application would be able to run on devices of all sorts.
I will try to learn how to implement this myself, but due to me not having an iPhone or access to one, I would only be able to support Android.

**Comment by edmirjano** [comment]
Still needed in 2023 @ionic-team

**Comment by end3r-man** [comment]
+1
